cve_bin_tool/checkers/binutils.py

```diff
@@ -72,66 +48,35 @@ def guess_contains(lines):
         # strings - Lists printable strings from files.
         r"can't set BFD default target to `%s': %s"
         # strip - Discards symbols.
-        # windmc - A Windows compatible message compiler.
+        # windmc - A Windows compatible message compiler.get_ver
         # windres - A compiler for Windows resource files.
     ]
-
-    for line in lines:
-        for signature in signatures:
-            pattern = re.compile(signature)
-            if pattern.search(line):
-                return 1
-    return 0
-
-
-def get_version(lines, filename):
-    """returns version information for binutiles as found in a given file.
-
-    VPkg: gnu, binutils
-    """
-    version_info = dict()
-
-    util_names = [
+    FILENAME_PATTERNS = [
+        # check if it's the library instead of the command line utils
+        r"libbfd-((\d+\.)*\d+)[\d\w.-]*so",
         # command line utils
-        "ld",  # the GNU linker.
-        "as",  # the GNU assembler.
-        "addr2line",  # Converts addresses into filenames and line numbers.
-        "ar",  # A utility for creating, modifying and extracting from archives.
-        "c++filt",  # Filter to demangle encoded C++ symbols.
-        "dlltool",  # Creates files for building and using DLLs.
-        "gold",  # A new, faster, ELF only linker, still in beta test.
-        "gprof",  # Displays profiling information.
-        "nlmconv",  # Converts object code into an NLM.
-        "nm",  # Lists symbols from object files.
-        "objcopy",  # Copies and translates object files.
-        "objdump",  # Displays information from object files.
-        "ranlib",  # Generates an index to the contents of an archive.
-        "readelf",  # Displays information from any ELF format object file.
-        "size",  # Lists the section sizes of an object or archive file.
-        "strings",  # Lists printable strings from files.
-        "strip",  # Discards symbols.
-        "windmc",  # A Windows compatible message compiler.
-        "windres",  # A compiler for Windows resource files.
+        r"ld",  # the GNU linker.
+        r"as",  # the GNU assembler.
+        r"addr2line",  # Converts addresses into filenames and line numbers.
+        r"ar",  # A utility for creating, modifying and extracting from archives.
+        r"c\+{2}filt",  # Filter to demangle encoded C++ symbols.
+        r"dlltool",  # Creates files for building and using DLLs.
+        r"gold",  # A new, faster, ELF only linker, still in beta test.
+        r"gprof",  # Displays profiling information.
+        r"nlmconv",  # Converts object code into an NLM.
+        r"nm",  # Lists symbols from object files.
+        r"objcopy",  # Copies and translates object files.
```

Comment:
Just FYI, you don't have to put `r` in front of the string unless you might have an issue with escape characters:

https://docs.python.org/3/library/re.html

> The solution is to use Python’s raw string notation for regular expression patterns; backslashes are not handled in any special way in a string literal prefixed with 'r'. So r"\n" is a two-character string containing '\' and 'n', while "\n" is a one-character string containing a newline. Usually patterns will be expressed in Python code using this raw string notation.